src/Cafe/OS/libs/nsyshid/Skylander.h

```diff
@@ -0,0 +1,98 @@
+#include <mutex>
+
+#include "nsyshid.h"
+#include "Backend.h"
+
+#include "Common/FileStream.h"
+
+namespace nsyshid
+{
+	class SkylanderPortalDevice final : public Device {
+	  public:
+		SkylanderPortalDevice();
+		~SkylanderPortalDevice() = default;
+
+		bool Open() override;
+
+		void Close() override;
+
+		bool IsOpened() override;
+
+		ReadResult Read(ReadMessage* message) override;
+
+		WriteResult Write(WriteMessage* message) override;
+
+		bool GetDescriptor(uint8 descType,
+						   uint8 descIndex,
+						   uint8 lang,
+						   uint8* output,
+						   uint32 outputMaxLength) override;
+
+		bool SetProtocol(uint32 ifIndex, uint32 protocol) override;
+
+		bool SetReport(ReportMessage* message) override;
+
+	  private:
+		bool m_IsOpened;
+	};
+
+	extern const std::map<const std::pair<const uint16, const uint16>, const std::string> listSkylanders;
```

Comment:
`listSkylanders` should be renamed to `s_listSkylanders` to match our naming convention. Even better would be to avoid using a loose global variable altogether. Instead, accessing it via a static getter method in SkylanderUSB would be preferable. What I mean is accessing it like this:
```
nsyshid::SkylanderUSB::GetSkylanders();
```
Which imho looks better than `nsyshid::skylanders`